elastic-apm-agent/src/test/java/co/elastic/apm/agent/premain/ShadedClassLoaderTest.java

```diff
@@ -36,20 +38,31 @@
 
 class ShadedClassLoaderTest {
 
+    @Nullable
+    private ShadedClassLoader cl = null;
+
+    @AfterEach
+    void cleanup() throws IOException {
+        // If classloader isn't properly closed it will make tests fail on Windows as the temporary folder
+        // can't be deleted, unlike on Linux/MacOS where files can be deleted from filesystem even if they are in use.
+        if (cl != null) {
+            cl.close();
+        }
```

Comment:
```suggestion
        cl.close();
```

[minor] any chance that `cl == null` at the end of a test, or is it only to remove the IDE warning? I think it's safer to fail on `NullPointerException`, so that if I add a test and test it on my mac, I cannot go along and merge it without storing the `ShadedClassLoader` reference in the dedicated field.

Reply:
If any test does not use the field then it can be `null`, however that would likely mean another potential failure on Windows, thus I'll add an explicit requirement to have it set with an NPE just to future-proof this :-).